src/sdo/sood/data/sdo_ml_v2_dataset.py

```diff
@@ -311,3 +327,58 @@ def predict_dataloader(self):
                           num_workers=self.num_workers,
                           pin_memory=self.pin_memory,
                           drop_last=self.drop_last,)
+
+
+def temporal_train_val_split(dataset: SDOMLv2NumpyDataset, split_ratio=0.7, temporal_chunk_size="6h",
```

Comment:
this will split the train and validation set into temporal chunks, feedback is welcome